app/util.py

```diff
@@ -386,32 +386,33 @@ def decorated(*args, **kwargs):
 
         return decorated
     
-    def verifyPasswordRules(f):
-        pattern = r'^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[!@#$%^&*()\-\+?_=,<>/]).{8,}$'
-        @wraps(f)               
-        def decorated(*args, **kwargs):
-            registerData = request.json
-            password = registerData.get("password", "")
-            email = registerData.get("email","")
-            username = registerData.get("username","")
-
-            if not email:
-                return errorResponse("Please provide an Email!", 400)
-
-            if not password:
-                return errorResponse("Password cannot be empty", 400)
-            
-            if len(password) < 8:
-                return errorResponse("Password needs to be at least 8 characters long")
-
-            if not re.match(pattern, password):
-                return errorResponse(
-                    " 👀 if u don't have at least one of a-z A-Z 0-9, and a special character (!@#$%^&*()-+?_=,<>/) , i keal u ", 
-                    400
+    def validatePassword(optional=False):
+        def decorator(f):
+            @wraps(f)
+            def decorated(*args, **kwargs):
+                data = request.json or {}
+                password = data.get("password")
+
+                if not password:
+                    if optional:
+                        return f(*args, **kwargs)
+                    return errorResponse("Password cannot be empty", 400)
+
+                if len(password) < 8:
+                    return errorResponse("Password needs to be at least 8 characters long", 400)
+
+                pattern = r'^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[!@#$%^&*()\-\+?_=,<>/]).{8,}$'
+                if not re.match(pattern, password):
+                    return errorResponse(
+                        "Password must include at least one of a-z, A-Z, 0-9, and a special character (!@#$%^&*()-+?_=,<>/).",
+                        400
                     )
 
-            if username.lower() in password.lower() or email.lower().split("@")[0] in password.lower():
-                return errorResponse("Your password should not contain your username or email!", 400)
+                username = data.get("username", "")
```

Comment:
what if the request body doesn't contain a username field, but the user chooses a password matching their username (in the db)?

check if `tokenPayload` exists in `g`, and in that case use `g.tokenPayload["user_id"]` to get the username & email from db